SC1058 false positive in bash for valid "for" loop without "do"

#### For bugs
- Rule Id (if any, e.g. SC1000): SC1058
- My shellcheck version (`shellcheck --version` or "online"): 0.7.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
for i in {1..10}; { printf '%s\n' "${i}"; }

```

#### Here's what shellcheck currently says:
```
for i in {1..10}; { printf '%s\n' "${i}"; }
^-- SC1073: Couldn't parse this for loop. Fix to allow more checks.
                  ^-- SC1058: Expected 'do'.
                  ^-- SC1072: Expected 'do'. Fix any mentioned problems and try again.
```

#### Here's what I wanted or expected to see:
No warning, in bash this syntax for for loops works.

--- a/src/ShellCheck/Parser.hs
+++ b/src/ShellCheck/Parser.hs
@@ -2448,6 +2448,7 @@ readDoGroup kwId = do
 
 
 prop_readForClause = isOk readForClause "for f in *; do rm \"$f\"; done"
+prop_readForClause1 = isOk readForClause "for f in *; { rm \"$f\"; }"
 prop_readForClause3 = isOk readForClause "for f; do foo; done"
 prop_readForClause4 = isOk readForClause "for((i=0; i<10; i++)); do echo $i; done"
 prop_readForClause5 = isOk readForClause "for ((i=0;i<10 && n>x;i++,--n))\ndo \necho $i\ndone"
@@ -2487,7 +2488,7 @@ readForClause = called "for loop" $ do
             "Don't use $ on the iterator name in for loops."
         name <- readVariableName `thenSkip` allspacing
         values <- readInClause <|> (optional readSequentialSep >> return [])
-        group <- readDoGroup id
+        group <- readBraced <|> readDoGroup id
         return $ T_ForIn id name values group
 
 prop_readSelectClause1 = isOk readSelectClause "select foo in *; do echo $foo; done"
@@ -3431,4 +3432,3 @@ tryWithErrors parser = do
 
 return []
 runTests = $quickCheckAll
-

